### PR TITLE
chore: CHANGELOG.md for v0.34.0

### PR DESCRIPTION
**Pull request by NagyZoltanPeter**
# Description
Changelog for v0.34.0

Please read through and add any highlight for this release as needed.

# Changes

- [x] changelog update

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,67 @@
...
+
+This release supports the following [libp2p protocols](https://docs.libp2p.io/concepts/protocols/):
+| Protocol | Spec status | Protocol id |
+| ---: | :---: | :--- |
+| [`11/WAKU2-RELAY`](https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/11/relay.md) | `stable` | `/vac/waku/relay/2.0.0` |
+| [`12/WAKU2-FILTER`](https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/12/filter.md) | `draft` | `/vac/waku/filter/2.0.0-beta1` <br />`/vac/waku/filter-subscribe/2.0.0-beta1` <br />`/vac/waku/filter-push/2.0.0-beta1` |
+| [`13/WAKU2-STORE`](https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/13/store.md) | `draft` | `/vac/waku/store/2.0.0-beta4` |
+| [`19/WAKU2-LIGHTPUSH`](https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/19/lightpush.md) | `draft` | `/vac/waku/lightpush/2.0.0-beta1` |
+| [`66/WAKU2-METADATA`](https://github.com/waku-org/specs/blob/master/standards/core/metadata.md) | `raw` | `/vac/waku/metadata/1.0.0` |
+| [`WAKU-SYNC`](https://github.com/waku-org/specs/blob/feat--waku-sync/standards/core/sync.md) | `draft` | `/vac/waku/sync/1.0.0` |
```
Comment (SionoiS) [inline CHANGELOG.md:63]: the spec was merged. new link ->
```suggestion
| [`WAKU-SYNC`](https://github.com/waku-org/specs/blob/master/standards/core/sync.md) | `draft` | `/vac/waku/sync/1.0.0` |
```

**Review by SionoiS — APPROVED** [review]
LGTM

**Comment by gabrielmer** [comment]
I just noticed that for 0.34 we have the milestone to remove the `protectedTopic` CLI flag https://github.com/waku-org/nwaku/milestone/22

I'll add a PR for it right now. Is it possible to add it to the release and include a note here in the changelog announcing that we removed it? 😶

File: CHANGELOG.md (hunk 1 above)
Comment (NagyZoltanPeter) [reply CHANGELOG.md:63]: Fixed it, thank you!

**Comment by NagyZoltanPeter** [comment]
> I just noticed that for 0.34 we have the milestone to remove the `protectedTopic` CLI flag https://github.com/waku-org/nwaku/milestone/22
> 
> I'll add a PR for it right now. Is it possible to add it to the release and include a note here in the changelog announcing that we removed it? 😶
> 
> update: this is the PR: #3160

I will cherry pick it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,66 @@
...
+- rejecting excess relay connections ([#3065](https://github.com/waku-org/nwaku/issues/3065)) ([8b0884c7](https://github.com/waku-org/nwaku/commit/8b0884c7))
+- static linking negentropy in ARM based mac ([#3046](https://github.com/waku-org/nwaku/issues/3046)) ([256b7853](https://github.com/waku-org/nwaku/commit/256b7853))
+
+### Changes
+
+- support ping with multiple multiaddresses and close stream ([#3154](https://github.com/waku-org/nwaku/issues/3154)) ([3665991a](https://github.com/waku-org/nwaku/commit/3665991a))
+- ** liteprotocoltester:** easy setup fleets for lpt ([#3125](https://github.com/waku-org/nwaku/issues/3125)) ([268e7e66](https://github.com/waku-org/nwaku/commit/268e7e66))
+- saving peers enr capabilities ([#3127](https://github.com/waku-org/nwaku/issues/3127)) ([69d9524f](https://github.com/waku-org/nwaku/commit/69d9524f))
+- **networkmonitor:** add missing field on RlnRelay init, set default for num of shard ([#3136](https://github.com/waku-org/nwaku/issues/3136)) ([edcb0e15](https://github.com/waku-org/nwaku/commit/edcb0e15))
```
Comment (Ivansete-status) [inline CHANGELOG.md:33]:
```suggestion
- networkmonitor: add missing field on RlnRelay init, set default for num of shard ([#3136](https://github.com/waku-org/nwaku/issues/3136)) ([edcb0e15](https://github.com/waku-org/nwaku/commit/edcb0e15))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,66 @@
+## v0.34.0 (2024-10-29)
+
+### Notes:
+
+* The `--protected-topic` CLI configuration has been removed. Equivalent flag, `--protected-shard`, shall be used instead.
+
+### Features
+
+- change latency buckets ([#3153](https://github.com/waku-org/nwaku/issues/3153)) ([956fde6e](https://github.com/waku-org/nwaku/commit/956fde6e))
+- **libwaku:** ping peer ([#3144](https://github.com/waku-org/nwaku/issues/3144)) ([de11e576](https://github.com/waku-org/nwaku/commit/de11e576))
```
Comment (Ivansete-status) [inline CHANGELOG.md:10]:
```suggestion
- libwaku: ping peer ([#3144](https://github.com/waku-org/nwaku/issues/3144)) ([de11e576](https://github.com/waku-org/nwaku/commit/de11e576))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,66 @@
...
+- initial windows support ([#3107](https://github.com/waku-org/nwaku/issues/3107)) ([ff21c01e](https://github.com/waku-org/nwaku/commit/ff21c01e))
+- Circuit relay ([#3112](https://github.com/waku-org/nwaku/issues/3112)) ([cfde7eea](https://github.com/waku-org/nwaku/commit/cfde7eea))
+
+### Bug Fixes
+
+- peer exchange libwaku response handling ([#3141](https://github.com/waku-org/nwaku/issues/3141)) ([76606421](https://github.com/waku-org/nwaku/commit/76606421))
+- add more logs, stagger intervals & set prune offset to 10% for waku sync ([#3142](https://github.com/waku-org/nwaku/issues/3142)) ([a386880b](https://github.com/waku-org/nwaku/commit/a386880b))
+- add log and archive message ingress for sync ([#3133](https://github.com/waku-org/nwaku/issues/3133)) ([80c7581a](https://github.com/waku-org/nwaku/commit/80c7581a))
+- add a limit of max 10 content topics per query ([#3117](https://github.com/waku-org/nwaku/issues/3117)) ([c35dc549](https://github.com/waku-org/nwaku/commit/c35dc549))
+- avoid segfault by setting a default num peers requested in PX ([#3122](https://github.com/waku-org/nwaku/issues/3122)) ([82fd5dde](https://github.com/waku-org/nwaku/commit/82fd5dde))
```
Comment (Ivansete-status) [inline CHANGELOG.md:20]:
```suggestion
- avoid segfault by setting a default num peers requested in Peer eXchange ([#3122](https://github.com/waku-org/nwaku/issues/3122)) ([82fd5dde](https://github.com/waku-org/nwaku/commit/82fd5dde))
```

**Review by Ivansete-status — COMMENTED** [review]
It looks great thanks!
I just added some minor comments.
We need to update the PR's title and description to properly reflect the underlying changes.
Not approving yet as per one question I ask re `pre-release-yml`.

**Review by gabrielmer — APPROVED** [review]
Thank you!

File: .github/workflows/pre-release.yml [hunk 5]
```diff
@@ -159,5 +159,5 @@ jobs:
           TARGET=$([[ "${{ needs.tag-name.outputs.tag }}" == "nightly" ]] && echo "--target ${{steps.vars.outputs.ref}}" || echo "")
 
           gh release create ${{ needs.tag-name.outputs.tag }} --prerelease ${TARGET} \
-            --title ${{ needs.tag-name.outputs.tag }} --notes-file release_notes.md \
-            wakunode2/* wakutools/*
```
[the remark this answers is not in the document]
Comment (NagyZoltanPeter) [reply .github/workflows/pre-release.yml]: Yes, change revoked. We need to fix the git hub action dependency rather.

**Pull request merged**